contentreview-impl/tool/src/java/org/sakaiproject/contentreview/servlet/SubmissionCallbackServlet.java

```diff
@@ -0,0 +1,113 @@
+package org.sakaiproject.contentreview.servlet;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONObject;
+
+import org.sakaiproject.assignment.api.AssignmentSubmissionEdit;
+import org.sakaiproject.assignment.cover.AssignmentService;
+import org.sakaiproject.component.cover.ComponentManager;
+import org.sakaiproject.content.cover.ContentHostingService;
+import org.sakaiproject.content.api.ContentResourceEdit;
+import org.sakaiproject.contentreview.model.ContentReviewItem;
+import org.sakaiproject.contentreview.service.ContentReviewService;
+import org.sakaiproject.entity.api.ResourcePropertiesEdit;
+import org.sakaiproject.event.cover.NotificationService;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.cover.SessionManager;
+
+/** 
+ * This servlet will receive callbacks from TII. Then it will process the data
+ * related to the submissions and resubmission and store it.
+ */
+
+@SuppressWarnings("deprecation")
+public class SubmissionCallbackServlet extends HttpServlet {
+	
+	private static Log M_log = LogFactory.getLog(SubmissionCallbackServlet.class);
+	
+	private ContentReviewService contentReviewService =
+				(ContentReviewService)ComponentManager.get("org.sakaiproject.contentreview.service.ContentReviewService");
+				
+	@Override
+	public void init(ServletConfig config) throws ServletException {
+		M_log.debug("init SubmissionCallbackServlet");
+		super.init(config);
+	}
+	
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		M_log.debug("doGet SubmissionCallbackServlet");
+		doPost(request, response);
+	}
+	
+	@SuppressWarnings("unchecked")
+	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		M_log.debug("doPost SubmissionCallbackServlet");
+		String contentType = request.getContentType();
+		if ( contentType != null && contentType.startsWith("application/json") ) {
+			doPostJSON(request, response);
+		} else {
+			M_log.warn("SubmissionCallbackServlet received a not json call.");
```

Comment:
In the log line we should indicate that the callback should have a Content-Type header of application/json to ease debugging.